Challenge2/app/src/main/java/com/bumble/challenge2/navmodel/custombackstack/transition/RecentsTransitionHandler.kt

```diff
@@ -48,7 +49,10 @@ class RecentsTransitionHandler<NavTarget>(
         when (this) {
             is State.Created -> created.copy(offset = Offset(0f, 2f * height))
             is State.Active -> active.copy(offset = activeOffset())
-            is State.Frozen -> TODO("(4) Frozen state should have blur of 15.dp. Note that blur only works Android 12+")
+            is State.Frozen -> active.copy() //I honestly have no idea how this should be implemented, as createModifier already seems to cover this..
+            //is State.Frozen -> createModifier(Modifier.blur(15.dp))
+            //is State.Frozen -> active.copy(specDp(15.dp))
+            //is State.Frozen -> TODO("(4) Frozen state should have blur of 15.dp. Note that blur only works Android 12+")
```

Comment:
In order to fix this you'd need to add a new property to `Props` called `blur: Dp = 0.dp`. 

Then define a field `frozen`, similar to `created`, `active` etc
`private val frozen = active.copy(blur = 15.dp)`

Then similarly to how we animate the other props, you'd add something like this inside `createModifier`
```
val blur by transition.animateDp(
            transitionSpec = specDp,
            targetValueByState = { it.toProps(height).blur },
            label = ""
        )
```

then apply this to your modifier:

```
this
            .offset {
                IntOffset(
                    x = (offset.x * density).roundToInt(),
                    y = (offset.y * density).roundToInt()
                )
            }
            .scale(scale)
            .graphicsLayer { rotationX = rotation }
            .drawBehind {
                drawRoundRect(
                    color = Color.Black,
                    cornerRadius = CornerRadius(x = 10f * density, y = 10f * density)
                )
            }
            .alpha(alpha)
            .blur(
                blur, edgeTreatment = BlurredEdgeTreatment(
                    // TODO this is in %, above it's defined in pixels
                    RoundedCornerShape(2)
                )
            )
            .zIndex(zIndex)
```